ext/storage_sources/s3_store/s3_log_system.cpp

```diff
@@ -0,0 +1,68 @@
+#include <aws/core/Aws.h>
+#include "s3_log_system.h"
+
+#include <cstdarg>
+
+S3LogSystem::S3LogSystem(WT_EXTENSION_API *wtApi, uint32_t wtVerbosityLevel)
+{
+    // Mapping the desired WiredTiger extension verbosity level to a rough equivalent AWS SDK
+    // verbosity level.
+    static const std::map<int32_t, Aws::Utils::Logging::LogLevel> verbosityMapping = {
+      {-3, Aws::Utils::Logging::LogLevel::Error}, {-2, Aws::Utils::Logging::LogLevel::Warn},
+      {-1, Aws::Utils::Logging::LogLevel::Info}, {0, Aws::Utils::Logging::LogLevel::Info},
+      {1, Aws::Utils::Logging::LogLevel::Debug}};
+    if (verbosityMapping.find(wtVerbosityLevel) != verbosityMapping.end()) {
+        logLevel = verbosityMapping.at(wtVerbosityLevel);
+    } else {
+        logLevel = Aws::Utils::Logging::LogLevel::Error;
+    }
+    this->wtApi = wtApi;
+}
+
+void
+S3LogSystem::Log(Aws::Utils::Logging::LogLevel logLevel, const char *tag, const char *format, ...)
+{
+    Aws::StringStream ss;
+    std::va_list args;
+    va_list tmpArgs; // unfortunately you cannot consume a va_list twice
+    va_start(args, format);
+
+#ifdef _WIN32
+    const int requiredLength = _vscprintf(formatStr, tmp_args) + 1;
+#else
+    const int requiredLength = vsnprintf(nullptr, 0, format, tmpArgs) + 1;
+#endif
+    va_end(tmpArgs);
+
+    Aws::Utils::Array<char> outputBuff(requiredLength);
```

Comment:
I do not know what ```Aws::Utils::Array<char> ``` is. Could we find an alternative from the standard library? My reading of this reflects, it could have been a char*